src/sentry/releasehealth/base.py

```diff
@@ -55,3 +60,43 @@ def get_current_and_previous_crash_free_rates(
             }
         """
         raise NotImplementedError()
+
+    class ReleaseAdoption(TypedDict):
+        #: Adoption rate (based on usercount) for a project's release from 0..100
+        adoption: Optional[float]
+        #: Adoption rate (based on sessioncount) for a project's release from 0..100
+        sessions_adoption: Optional[float]
+        #: User count for a project's release (past 24h)
+        users_24h: Optional[int]
+        #: Sessions count for a project's release (past 24h)
+        sessions_24h: Optional[int]
+        #: Sessions count for the entire project (past 24h)
+        project_users_24h: Optional[int]
+        #: Sessions count for the entire project (past 24h)
+        project_sessions_24h: Optional[int]
+
+    ReleasesAdoption = Mapping[Tuple[ProjectId, ReleaseName], ReleaseAdoption]
+
+    def get_release_adoption(
+        self,
+        project_releases: Sequence[Tuple[ProjectId, ReleaseName]],
+        environments: Optional[Sequence[EnvironmentName]] = None,
+        now: Optional[datetime] = None,
+        org_id: Optional[OrganizationId] = None,
+    ) -> ReleasesAdoption:
+        """
+        Get the adoption of the last 24 hours (or a difference reference timestamp).
+
+        :param project_releases: A list of releases to get adoption for. Our
+            backends store session data per-project, so each release has to be
+            scoped down to a project too.
+
+        :param environments: Optional. A list of environments to filter by.
+        :param now: Release adoption information will be provided from 24h ago
+            until this timestamp.
+        :param org_id: An organization ID to filter by. Note that all projects
+            have to be within this organization, and this backend doesn't check for
+            that. Omit if you're not sure.
+        """
+
+        raise NotImplementedError()
```

Comment:
Nit: I would add an error message for whoever will subclass this in the future to know where this is coming from